### DISCUSSION TITLE
Possible to use starship with a brand new shell, or does it require something on the starship side first?

### DISCUSSION THREAD

**Discussion by MostHated** [starship/starship#2905] [Q&A] [answered]
Hey there,
    I wasn't sure if I should be asking here, or to the developer of the shell as to compatibility, but I figured this was the best place to start. I have been playing around with the [dotnet-shell](https://github.com/dotnet-shell/Shell) on Pop!_OS 21.04, but would like to be able to use starship with it if possible. It allows for relatively easy prompt customization already, which made me wonder if starship could work with minimal effort as well?

Here is where the prompt is currently set, in the equivalent of a .bashrc [defaults/core](https://github.com/dotnet-shell/Shell/blob/b30deaf075ea4c32fd23829243984c191ed7f1b6/src/FirstRunWizard/defaults/core#L13)

The usage seems pretty similar for most shells, but the type of shell is passed in after the init (which I am guessing is partially to let it know which type and where to locate the config?), so since this one is brand new, I would imagine that passing anything in would do no good unless it exists in starship already. If anyone has any thoughts on this, be it a simple yeah or nay that it would be possible, I would appreciate it.

Thanks,
-MH

**Comment by matchai** [comment]
Hey @MostHated,

I'd recommend reading over [this FAQ entry](https://starship.rs/faq/#the-docs-say-starship-is-cross-shell-why-isn-t-my-preferred-shell-supported).
In short, it should be possible to quickly write your own wrapper for Starship in most shells with command expansion.
You can take a look at the files in [`src/init`](https://github.com/starship/starship/tree/master/src/init) for inspiration.

**Comment by i-am-shodan** [comment, marked as answer]
Done a quick test and replacing the existing prompt handler with this seems to work fine.

```
Shell.Prompt = () =>
    var prompt = `starship prompt --status=$Shell.LastExitCode$ --jobs=$Shell.BackgroundProcesses.Count()$`;
    return ColorString.FromRawANSI(prompt);
};
```
